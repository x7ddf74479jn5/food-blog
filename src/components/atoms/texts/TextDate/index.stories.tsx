import type { ComponentMeta, ComponentStory } from "@storybook/react";

import TextDate from ".";

export default {
  argTypes: {
    date: { control: { type: "date" } },
  },
  component: TextDate,
  title: "Atoms/texts/TextDate",
} as ComponentMeta<typeof TextDate>;

const Template: ComponentStory<typeof TextDate> = (args) => <TextDate {...args} />;

export const Default = Template.bind({});
Default.args = {
  date: new Date("2020-01-01"),
};
