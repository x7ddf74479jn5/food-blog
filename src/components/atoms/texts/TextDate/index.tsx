import { formatISO } from "date-fns";

import { formatJpYYYYMD } from "@/lib/date";

type Props = {
  date: Date;
};

export const TextDate = ({ date }: Props) => {
  return (
    <time dateTime={formatISO(date)}>
      <span>{formatJpYYYYMD(date)}</span>
    </time>
  );
};

export default TextDate;
