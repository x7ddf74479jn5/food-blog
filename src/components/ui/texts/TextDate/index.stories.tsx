import type { ComponentMeta, ComponentStoryObj } from "@storybook/react";

import TextDate from ".";

export default {
  argTypes: {
    date: { control: { type: "date" } },
  },
  component: TextDate,
  title: "ui/texts/TextDate",
} as ComponentMeta<typeof TextDate>;

export const Default: ComponentStoryObj<typeof TextDate> = {
  args: {
    date: new Date("2020-01-01"),
  },
};
