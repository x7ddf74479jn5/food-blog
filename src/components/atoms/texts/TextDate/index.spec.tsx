import renderer from "react-test-renderer";

import TextDate from ".";

describe("components/atoms/TextDate", () => {
  const date = new Date("2020-12-12T00:00:00+09:00");
  it("snapshot", () => {
    const tree = renderer.create(<TextDate date={date} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
