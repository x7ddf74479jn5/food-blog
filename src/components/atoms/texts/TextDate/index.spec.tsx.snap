// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/TextDate snapshot 1`] = `
<time
  dateTime="2020-12-12T00:00:00+09:00"
>
  <span>
    2020年12月12日
  </span>
</time>
`;
